mapboxgl/templates/vector_choropleth.html

```diff
@@ -0,0 +1,104 @@
+{% extends "choropleth.html" %}
+    
+    {% block choropleth %}
+
+    var joinData = {{ joinData }};
+
+    {% if joinData %}
+        var vectorColorStops = [],
+            popUpKeys = [];
+
+        // create categorical color stops for features in vector layer
+        joinData.forEach(function(row, index) {
+            var red = Math.max(255 - row["{{ colorProperty }}"], 0),
+                green = Math.max(255 - row["{{ colorProperty }}"], 0),
+                blue = Math.min(204 + row["{{ colorProperty }}"], 255);
+
+            var color = "rgba(" + red + ", " + green + ", " + blue + ", 1)";
+            vectorColorStops.push([row["id"], color]);
+            
+            // for adding joined data to pop-up
+            popUpKeys[row["{{ dataJoinProperty }}"]] = row["{{ colorProperty }}"];
+        });
+    {% endif %}
+
+    // Add vector data source
+    map.addSource("vector-data", {
+        type: "vector",
+        url: "{{ vectorUrl }}",
+    });
+
+    // Add layer from the vector tile source with data-driven style
+    map.addLayer({
+        "id": "choropleth-fill",
+        "type": "fill",
+        "source": "vector-data",
+        "source-layer": "{{ vectorLayer }}",
+        "paint": {
+            "fill-color": generatePropertyExpression("match", "{{ vectorJoinColorProperty }}", vectorColorStops, "{{ defaultColor }}"), 
+            "fill-opacity": {{ opacity }}
+        }
+    }, "{{ belowLayer }}");
+
+    // Add border layer
+    map.addLayer({
+        "id": "choropleth-line",
+        "source": "vector-data",
+        "source-layer": "{{ vectorLayer }}",
+        "type": "line",
+        "layout": {
+            "line-join": "round",
+            "line-cap": "round"
+        },
+        "paint": {
+            {% if lineDashArray %}
+                "line-dasharray": {{ lineDashArray }},
+            {% endif %}
+            "line-color": "{{ lineColor }}",
+            "line-width": {{ lineWidth }},
+            "line-opacity": {{ opacity }}
+        }
+    }, "{{ belowLayer }}" );
+
+    // Add label layer
+    map.addLayer({
+        "id": "choropleth-label",
+        "source": "vector-data",
+        "source-layer": "{{ vectorLayer }}",
+        "type": "symbol",
+        "layout": {
+            {% if labelProperty %}
+            "text-field": "{{ labelProperty }}",
+            {% endif %}
+            "text-size" : generateInterpolateExpression('zoom', [[0,8],[22,16]] ),
+            "text-offset": [0,-1]
+        },
+        "paint": {
+            "text-halo-color": "white",
+            "text-halo-width": 1
+        }
+    }, "{{belowLayer}}" );
+
+    {% endblock choropleth %}
+
+    {% block choropleth_popup %}
```

Comment:
When I run this example code, the mousemove function below is not applied to the map I believe because the `choropleth_popup` template in `choropleth.html` is being executed instead of this block.  I'm sure this could be addressed multiple ways, but I confirmed a quick fix by adding the same block name `choropleth_popup` to both `choropleth.html` and `vector_choropleth.html` with the complete code for the popup specific to that template in each.

Reply:
I think I've covered this in 722d747, but I wasn't able to replicate the original problem.  Can you confirm if it's working now?